PRFCS/prfc-1.md

```diff
@@ -51,50 +54,50 @@ Returns information about the Token implemented by this contract.
 fn allowance(owner: PublicAddress, spender: PublicAddress) -> u64
 ```
 
-Returns the number of tokens currently in the spender's allowance that they can spend on the ‘owner’s’ behalf.
+Returns the amount of tokens that the `spender` can currently spend on behalf of the `owner`.
 
 ### balance_of
 ```rust
 fn balance_of(address: PublicAddress) -> u64
 ```
 
-Queries the balance for an owner account 'address'.
+Queries the amount of tokens in an account identified by `address`.
 
 
 ## Required Calls
 ---
 
 ### transfer
 ```rust
-fn transfer(to_address: Option<Public_Address>, amount: u64)
+fn transfer(to_address: Option<PublicAddress>, amount: u64)
 ```
 
-`transfer` transfers tokens to an address from the account identified by `txn.signer`. If `to_address` is None, this burns the amount.
+Transfers tokens to an address from the account identified by `calling_account`. If `to_address` is None, this burns the amount.
 
-`transfer` must panic if `txn.signer`'s balance is less than the amount.
+`transfer` must panic if `balance_of(calling_account)` < `amount`.
 
-Log `Transfer` must be emitted if 'transfer' is successful.
+Log `Transfer` must be emitted if `transfer` is successful.
 
 
 ### transfer_from
 ```rust
-fn transfer_from(from_address: PublicAddress, to_address: Option<PublicAddress>, value: u64)
+fn transfer_from(from_address: PublicAddress, to_address: Option<PublicAddress>, amount: u64)
 ```
 
-`transfer_from` transfers tokens to an address on behalf of the ‘owner’. If `to_address` is None, this burns the amount.
+Transfers tokens to an address on behalf of the owner (`from_address`). If `to_address` is None, this burns the amount.
 
-`transfer_from` must panic if get_allowance_from(owner) < value.
+`transfer_from` must panic if `allowance(from_address, calling_account)` < `amount`.
 
-Log `Transfer` must trigger if `transfer_from` is successful. Note that the value of the `Transfer` log must contain the owner's address, not `txn.signer`.
+Log `Transfer` must trigger if `transfer_from` is successful. Note that the topic of the `Transfer` log must contain the owner's address, not `calling_account`.
```

Comment:
Log `Transfer` must **be emitted** if `transfer_from` is successful. Note that the topic of the `Transfer` log must contain the owner's address (`from_address`), not the address of the `calling_account`.